### PR TITLE
Let Prowlarr use TvdbIds when searching for tv releases on TorrentSyndikat

### PR DESCRIPTION
**Pull request by stiivo**
Let Prowlarr use TorrentSyndikat API with TVDbIds

If theres a tvdbid the searchterm is omitted

#### Database Migration
NO

#### Description
These changes allow Prowlarr to use TVDB-IDs when search with the Torrent-Syndikat Indexer

#### Todos
- [x] Tests
- [ ] Translation Keys (./src/NzbDrone.Core/Localization/Core/en.json)
- [ ] [Wiki Updates](https://wiki.servarr.com)

### CODE REVIEW

**Comment by bakerboy448** [comment]
If it's only searching TVDBid won't then return all results for the entire series and not the season nor season/episode being looked for?

**Comment by stiivo** [comment]
Good point, thank you. 

I will rework that. The point is, the API is not cappable of searching for international titles. To get correct results for example:
Married... with Children S11 i have to ommit the english title and use the tvdbid instead OR get Sonarr/Prowlarr to use german titles which looks like the harder way for me. I will change the code so that if existing, only the Season/Episode marker lands in the searchstring.

Does that sound right to you?

**Comment by bakerboy448** [comment]
Yup! 

And that works easier than needing to get sonarr to add a ton of aliases of course :D

**Comment by stiivo** [comment]
Turns out that when the searchstring parameter of the Tracker API is used, all other parameters such as tvbId are ignored. I am in contact with the admin to get this fixed.

**Comment by stiivo** [comment]
Unfortunately the admin is not willing to make the necessary changes. He thinks it is better if the evaluation of the results is done on the client side.

**Comment by bakerboy448** [comment]
Do they drop individual episodes when seasons end and only have season packs?

Regardless \*arrs will only page back up to 1000 results. So any series that has more than 1000 results by id, that logic isn't valid.

**Comment by stiivo** [comment]
Yes, it seems that single episodes are replaced with a season pack when available.

I also tried to explain that narrowing down the possible results based on the tvdbId should rather help performance. But the argument was obviously not good enough. 

Annoying...

**Comment by stiivo** [comment]
Would it be better to create another profile for this tracker instead of editing the existing one? You could then just run both indexer profiles. I have no idea what side effects the current changes have for other users.

**Comment by stiivo** [comment]
@bakerboy448 could you please have a look on the last commit. I hope i found a way to ensure backwards compatibility to keep things working for others but also made it possible to make use of tvdbids when preferred.

**Review by bakerboy448 — CHANGES_REQUESTED** [review]
also defer to @Qstick on the implementation as well.
TL;DR add TVDBId support, but the site does not support any other params if TVDBId is used.

FWIW for similar sites, we have kept tvdbid search removed rather than jumping through hoops to add an option.....because no matter what as TVDBId is reported as supported, it will be attempted to be used by Sonarr.

File: src/NzbDrone.Core/Indexers/Definitions/TorrentSyndikat.cs [hunk 1]
```diff
@@ -184,7 +188,7 @@ public IndexerPageableRequestChain GetSearchRequests(TvSearchCriteria searchCrit
         {
             var pageableRequests = new IndexerPageableRequestChain();
 
-            pageableRequests.Add(GetPagedRequests(string.Format("{0}", searchCriteria.SanitizedTvSearchString), searchCriteria.Categories, searchCriteria.ImdbId));
+            pageableRequests.Add(GetPagedRequests(string.Format("{0}", searchCriteria.SanitizedSearchTerm), searchCriteria.Categories, searchCriteria.ImdbId, searchCriteria.TvdbId));
```
[the remark this answers is not in the document]
Comment (stiivo) [reply src/NzbDrone.Core/Indexers/Definitions/TorrentSyndikat.cs:191]: Well, that was definitly not intended.

File: src/NzbDrone.Core/Indexers/Definitions/TorrentSyndikat.cs [hunk 2]
```diff
@@ -141,7 +141,11 @@ private IEnumerable<IndexerRequest> GetPagedRequests(string term, int[] categori
             {
                 queryCollection.Add("imdbId", imdbId);
             }
-            else if (!string.IsNullOrWhiteSpace(searchString))
+            else if (tvdbId != null && (Settings.PreferTvDbId || string.IsNullOrWhiteSpace(searchString)))
```
[the remark this answers is not in the document]
Comment (stiivo) [reply src/NzbDrone.Core/Indexers/Definitions/TorrentSyndikat.cs:144]: Okay, i think i got it now.

> Basically in this case using the ID only - as I alluded to earlier - would return every release for the series on the site and Sonarr only pages back up to 1,000 results. so for any and all series with more than 1,000 releases on the site when searched by ID....sonarr would never be able to find those releases.

That is indeed the case for a total of 2 shows on this tracker. 

> so having it as an option - unless no results are returned if the option is not selected somehow - isn't feasible.

It depends, since its a german tracker, mainly german titles / release names are used. As i stated earlier in my example there are 0 results when searching for "Married with Children".

In that specific case, for this tracker, i have the feeling that the TVDBId search is the lesser evil because everytime one calls the API with a english title which differs from the german title there are 0 results.
